IPPO ready for launch

prateekbhustali opened the pull request:
+ updated metrics to follow Q-Learning algs for consistency
+ created IPPO sweep config
+ updated default IPPO config
+ added model checkpointing
+ added utility function `load_agent_checkpoint`
+ eval metrics also stored in an array. This allows us to conveniently retrieve `evals_returns` when vmapping over seeds. This also allows us stored them in wandb summary for quick retrieval*

Some example IPPO runs [here](https://wandb.ai/imp-act/jaxMARL-ippo_ToyExample-v2/workspace?nw=nwusershaktimaan)

Default hypers have good performance:

<img width="399" alt="image" src="https://github.com/user-attachments/assets/aa483d8e-f486-409d-8d2d-39732d50fef8" />

* It works well when summary has 10 entries:
![image](https://github.com/user-attachments/assets/e1d0fb35-c9ec-4dc3-93b3-30f70e4c0e90)

* But it doesn't work when the summary has 100 entries:
<img width="265" alt="image" src="https://github.com/user-attachments/assets/595ca8ed-bfd7-4df2-80b9-74d5ab3fbf93" />

prateekbhustali committed:
A copy of ippo_rnn_smax.py and its config, renamed to road_env.yaml
Creating this first so it is easier to track changes

prateekbhustali committed:
Remove ippo_ff_road_env.py

prateekbhustali committed:
initial commit ippo_rnn_road_env.py

prateekbhustali committed:
Minor formatting improvements

prateekbhustali committed:
Make NUM_UPDATES primary specification, and TOTAL_TIMESTEPS are now inferred

prateekbhustali committed:
Log metrics similar to Q-Learning, and also save eval metrics in summary

prateekbhustali committed:
Remove double quotes from config

prateekbhustali committed:
Logging GPU stats

prateekbhustali committed:
Add model checkpointing and model loading

prateekbhustali committed:
Make gpu stats logging identical for consistency

prateekbhustali committed:
Merge remote-tracking branch 'origin/extra_observations' into launch-checklist

prateekbhustali committed:
Update subproject commit reference in imp-act

prateekbhustali committed:
Update subproject commit reference in imp-act

prateekbhustali committed:
Merge branch 'imp_act_adaption' into launch-checklist

prateekbhustali committed:
Merge remote-tracking branch 'origin/hyper_parameter_tuning' into launch-checklist

prateekbhustali committed:
Merge remote-tracking branch 'origin/hyper_parameter_tuning' into launch-checklist

prateekbhustali committed:
Update variable `world_state_size`  to include extra_observations

prateekbhustali committed:
Update MAPPO `single_run(config)` according to Q-Learning for consistency

prateekbhustali committed:
Create MAPPO sweep and update `single_run(config)`

prateekbhustali committed:
Merge branch 'launch-checklist' into ippo-rnn

prateekbhustali committed:
remove unused plotting funcionality

prateekbhustali committed:
Log metrics similar to Q-Learning, and also save eval metrics in summary

prateekbhustali committed:
Make NUM_UPDATES primary specification, and TOTAL_TIMESTEPS are now inferred

prateekbhustali committed:
update config to include env extras and DOUBLE_PRECISION_MODE

prateekbhustali committed:
single_run now allows config overrides

prateekbhustali committed:
Add model checkpointing and model loading

prateekbhustali committed:
Remove unnecessary 'IF_SAVE' for model checkpointing

prateekbhustali committed:
Remove 'test' in sweep name

prateekbhustali committed:
Merge branch 'launch-checklist' into ippo-rnn

prateekbhustali committed:
Remove unnecessary 'IF_SAVE' for model checkpointing

prateekbhustali committed:
add IPPO sweep and config in /config/hyper_parameter_tuning/toy_example_v2

prateekbhustali committed:
Remove extra config file, and update default hypers

moratodpg commented on getting-started/config/hyper_parameter_tuning/toy_example_v2/ippo_sweep.yaml:
```diff
@@ -0,0 +1,67 @@
+  - ${env}
+  - python
+  - ${program} 
+  - --config-path=./config/hyper_parameter_tuning/toy_example_v2
```
Are we missing the default ippo yaml file in this folder?

prateekbhustali replied:
yes, thanks for pointing out. I've added it now!

moratodpg reviewed (approved):
IPPO looks great, @omniscientoctopus !

A few comments from my side:
+ I think an ippo config file is missing in the hyperparameter_tuning folder 
+ All seeds are named equally on wandb 
<img width="367" alt="image" src="https://github.com/user-attachments/assets/955c3021-70ed-4479-83fc-7844e3d00a97" />

+ (Minor issue) Configs are not formatted with quotes (different from the other algs)

+ (Question) Will we have problems with hydra if we run 2 simulations simultaneously (the same second)?
![image](https://github.com/user-attachments/assets/2d7424d0-eadf-4351-89c4-3e1435b61cfa)

prateekbhustali committed:
Update formatting of ippo_rnn_road_env.yaml for consistency

prateekbhustali committed:
add default IPPO yaml in sweep

prateekbhustali commented:
> All seeds are named equally on wandb

It was because "SEED" was set to 0 in the default config. If I set it to "random", each run name will be unique. I have changed it now.

> Configs are not formatted with quotes (different from the other algs)

I have fixed this ✅

> Will we have problems with hydra if we run 2 simulations simultaneously (the same second)?

I am not sure about this 🤔. In any case, config will be logged to wandb, so we can disregard this local copy?

DanielHettegger commented on getting-started/ippo_rnn_road_env.py:
```diff
@@ -0,0 +1,838 @@
+        rnn_state = carry
+        ins, resets = x
+
+        # ToDO: Remove after fixing env
```
I think those comments were removed in the MAPPO file

prateekbhustali replied:
resolved, thanks!

DanielHettegger commented on getting-started/ippo_rnn_road_env.py:
```diff
@@ -0,0 +1,838 @@
+        config["SEED"] = np.random.randint(0, 2**32 - 1)
+
+    wandb.run.name = f"{alg_name}_{env_name}_{map_name}_{config['SEED']}"
+    wandb.run.save()
```
This call seems to be redundant. I got it from some old wandb docs and now we have them in all scripts - sorry :sweat_smile:. wand.run.save() can be safely removed from all the scripts

prateekbhustali replied:
I will keept it for now, we can remove all of them at once after the merge?

DanielHettegger reviewed (approved):
Looks great! Thanks Prateek! Just some minor comments from my side.

prateekbhustali committed:
Remove ToDO comment
